packages/rest/src/lib/utils/types.ts

```diff
@@ -155,10 +155,20 @@ export interface RateLimitData {
 	 * The HTTP method being performed
 	 */
 	method: string;
+	/**
+	 * The time, in milliseconds, that will need to pass before this request can be retried
+	 */
+	retryAfter: number;
 	/**
 	 * The route being hit in this request
 	 */
 	route: string;
+	/**
+	 * The time, in milliseconds, that will need to pass before this request can be retried
+	 *
+	 * This is only present on certain sublimits, and `0` otherwise
+	 */
+	sublimitTimeout: number;
```

Comment:
why `0` and not null for non-existent?

Reply:
Mostly for consistency (and because it will always be present when called during a 429 encounter, but never present before the request gets sent because sublimits are already handled)

Reply:
We could even make it be same as retryAfter in pre-request onRatelimited scenario (which then would make it equal to post-429 scenario)

Reply:
leaving as is for this. Further sublimited events *should* emit with this (and possibly emit at queue time), but currently do not. Will need to be fixed in a follwup PR later.